src/BuiltInTools/BrowserRefresh/BrowserRefreshMiddleware.cs

```diff
@@ -59,6 +70,86 @@ public async Task InvokeAsync(HttpContext context)
             }
         }
 
+        private void AttachWebAssemblyHeaders(HttpContext context)
+        {
+            context.Response.OnStarting(() =>
+            {
+                if (!context.Response.Headers.ContainsKey("DOTNET-MODIFIABLE-ASSEMBLIES"))
+                {
+                    if(_dotnetModifiableAssemblies != null)
+                    {
+                        context.Response.Headers.Add("DOTNET-MODIFIABLE-ASSEMBLIES", _dotnetModifiableAssemblies);
+                    }
+                    else
+                    {
+                        _logger.LogDebug("DOTNET_MODIFIABLE_ASSEMBLIES environment variable is not set. The browser refresh feature may not work as expected.");
```

Comment:
This will log even when hot reload is explicitly disabled, right? Maybe we could explain this in the log message so customers don't automatically assume there's a problem (and report bugs on it, etc.). Something like:

```suggestion
                        _logger.LogDebug("DOTNET_MODIFIABLE_ASSEMBLIES environment variable is not set, likely because hot reload is not enabled. The browser refresh feature may not work as expected.");
```